File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.Arrays;
import java.util.List;

class ResistorColorTrio {

    static private List<String> colorBands = Arrays.asList("black", "brown", "red", "orange", "yellow", "green", "blue",
            "violet", "grey", "white");

    String label(String[] colors) {
        long resistance = colorBands.indexOf(colors[0]) * 10 + colorBands.indexOf(colors[1]);
        resistance *= (int) Math.pow(10, colorBands.indexOf(colors[2]));

        if (resistance >= 1000000000) {
            return String.format("%d gigaohms", resistance / 1000000000);
        } else if (resistance >= 1000000) {
            return String.format("%d megaohms", resistance / 1000000);
        } else if (resistance >= 1000) {
            return String.format("%d kiloohms", resistance / 1000);
        }

        System.out.println(resistance);

        return String.format("%d ohms", resistance);
    }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"]

def label(colors):
    match (COLORS.index(colors[0]) * 10 + COLORS.index(colors[1])) * (10 ** COLORS.index(colors[2])):
        case n if n > 10 ** 9: return f"{n//10**9} gigaohms"
        case n if n > 10 ** 6: return f"{n//10**6} megaohms"
        case n if n > 10 ** 3: return f"{n//10**3} kiloohms"
        case n:                return f"{n} ohms"
